### PR TITLE
refactor(store): Refactor reducer initialisation

### PR DESCRIPTION
**Pull request by ert78gb**
Refactored the ngrx/store reducer initialisation, because hard to extend the original solution.
Bad practise the object initialisation inside the components / services. The new solution use angular DI everywhere.
Separated the web and electron configuration store.

### CODE REVIEW

File: shared/src/services/datastorage-repository.service.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { InjectionToken } from '@angular/core';
+
+import { UserConfiguration } from '../config-serializer/config-items/UserConfiguration';
+
+export interface IDataStorageRepositoryService {
```
Comment (fjozsef) [inline shared/src/services/datastorage-repository.service.ts]: Consider renaming the interface and remove 'I'.
https://angular.io/styleguide

File: shared/src/store/effects/keymap.ts [hunk 2]
```diff
@@ -2,11 +2,13 @@ import { Injectable } from '@angular/core';
...
 
 import { Actions, Effect } from '@ngrx/effects';
-import { Store } from '@ngrx/store';
+import { Action, Store } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
 
 import 'rxjs/add/operator/do';
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/withLatestFrom';
+import 'rxjs/observable/of';
```
Comment (fjozsef) [inline shared/src/store/effects/keymap.ts]: Change to 'rxjs/add/observable/of'.
startWith and switchMap operators also need to be imported.

File: shared/src/store/effects/user-config.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Injectable, Inject } from '@angular/core';
+import { Effect, Actions } from '@ngrx/effects';
+import { Observable } from 'rxjs/Observable';
+import { Action } from '@ngrx/store';
+
+import { ActionTypes, LoadUserConfigAction, LoadUserConfigSuccessAction } from '../actions/user-config';
+import { UserConfiguration } from '../../config-serializer/config-items/UserConfiguration';
+import { IDataStorageRepositoryService, DATA_STORAGE_REPOSITORY } from '../../services/datastorage-repository.service';
+import { DefaultUserConfigurationService } from '../../services/default-user-configuration.service';
+
```
Comment (fjozsef) [inline shared/src/store/effects/user-config.ts]: Missing operator imports

File: shared/src/services/local-datastorage-repository.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { UserConfiguration } from '../config-serializer/config-items/UserConfiguration';
+import { IDataStorageRepositoryService } from './datastorage-repository.service';
+import {DefaultUserConfigurationService} from './default-user-configuration.service';
```
Comment (fjozsef) [inline shared/src/services/local-datastorage-repository.service.ts]: Add space after { and before {.

File: shared/src/services/datastorage-repository.service.ts (hunk 1 above)
Comment (ert78gb) [inline shared/src/services/datastorage-repository.service.ts]: I removed

File: shared/src/store/effects/keymap.ts (hunk 2 above)
Comment (ert78gb) [inline shared/src/store/effects/keymap.ts]: Added and changed the of

File: shared/src/store/effects/user-config.ts (hunk 3 above)
Comment (ert78gb) [inline shared/src/store/effects/user-config.ts]: Added missing operators

File: shared/src/services/local-datastorage-repository.service.ts (hunk 4 above)
Comment (ert78gb) [inline shared/src/services/local-datastorage-repository.service.ts]: added spaces

**Comment by fjozsef** [comment]
@ert78gb Could you rebase instead of merge? It is very annoying that I can see untouched codes from master.

**Comment by ert78gb** [comment]
I rebased. My PR method:
- update local master / from UHK master
- push local master -> github master
- checkout feature branch
- rebase to master

this is the result. SRY It is confuse me too. If you could better solution I accept it.

**Comment by fjozsef** [comment]
I think your process is fine.

**Pull request merged**